lms/static/scripts/postmessage_json_rpc/server.js

```diff
@@ -112,6 +120,39 @@ export class Server {
     /** @type {WindowProxy} */ (event.source).postMessage(result, event.origin);
   }
 
+  /**
+   * JSON-RPC 2.0 notification requests are distinguished by the lack of an
+   * `id` property.
+   *
+   * @param {MessageEvent} event
+   */
+  _isJSONRPCNotificationRequest(event) {
+    return typeof event.data.id === 'undefined';
+  }
+
+  /**
+   * Invoke the method indicated in the JSON-RPC 2.0 notification request
+   * object.
+   *
+   * @param {JsonRpcNotificationRequest} request
+   */
+  _jsonRPCNotify(request) {
+    const method = this._registeredMethods[request.method];
+
+    if (!method) {
+      console.error(
+        `JSON RPC notification for unrecognized method: ${request.method}`
+      );
+      return;
+    }
+
+    try {
+      method(...(request.params ?? []));
+    } catch (e) {
+      console.error(`JSON RPC notification method failed: ${e}`);
```

Comment:
For the `reportActivity` event handler, are you planning to catch that in the handler itself and show an error? As I mentioned on the call early this week, I think it is important to make the user aware if something goes wrong in the LMS app reporting to the backend, since this will cause the user's work not to show up in the grading view.

Reply:
This is important to do, but outside of the scope of this set of changes.

Reply:
https://github.com/hypothesis/lms/issues/3938